packages/desktop-client/src/components/common/Button2.tsx

```diff
@@ -146,59 +154,45 @@ export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
 
-    const hoveredStyle = {
-      ...(variant !== 'bare' && styles.shadow),
-      backgroundColor: backgroundColorHover[variant],
-      color: textColorHover[variant],
-      cursor: 'pointer',
-    };
-    const activeStyle = {
-      ..._getActiveStyles(variant, bounce),
-    };
-
-    const defaultButtonClassName: ReactAriaButtonClassNameFn = renderProps =>
-      String(
-        css({
-          alignItems: 'center',
-          justifyContent: 'center',
-          flexShrink: 0,
-          padding: _getPadding(variant),
-          margin: 0,
-          overflow: 'hidden',
-          display: 'flex',
-          borderRadius: 4,
-          backgroundColor: backgroundColor[variantWithDisabled],
-          border: _getBorder(variant, variantWithDisabled),
-          color: textColor[variantWithDisabled],
-          transition: 'box-shadow .25s',
-          WebkitAppRegion: 'no-drag',
-          ...styles.smallText,
-          ...(renderProps.isDisabled
-            ? {}
-            : {
-                '&[data-hovered]': hoveredStyle,
-                '&[data-pressed]': activeStyle,
-              }),
-          ...(Icon ? { paddingLeft: 0 } : {}),
-        }),
-      );
+    const defaultButtonClassName: string = useMemo(
+      () =>
+        String(
+          css({
+            alignItems: 'center',
+            justifyContent: 'center',
+            flexShrink: 0,
+            padding: _getPadding(variant),
+            margin: 0,
+            overflow: 'hidden',
+            display: 'flex',
+            borderRadius: 4,
+            backgroundColor: backgroundColor[variantWithDisabled],
+            border: _getBorder(variant, variantWithDisabled),
+            color: textColor[variantWithDisabled],
+            transition: 'box-shadow .25s',
+            WebkitAppRegion: 'no-drag',
+            ...styles.smallText,
+            '&[data-hovered]': _getHoveredStyles(variant),
+            '&[data-pressed]': _getActiveStyles(variant, bounce),
+            ...(Icon ? { paddingLeft: 0 } : {}),
+          }),
+        ),
+      [Icon, bounce, variant, variantWithDisabled],
+    );
 
-    const buttonClassName: ReactAriaButtonClassNameFn = renderProps =>
-      typeof props.className === 'function'
-        ? props.className(renderProps)
-        : props.className || '';
+    const className = restProps.className;
 
     return (
       <ReactAriaButton
         ref={ref}
         {...restProps}
-        className={renderProps =>
-          `${renderProps.defaultClassName} ${defaultButtonClassName(renderProps)} ${buttonClassName(renderProps)}`
+        className={
+          typeof className === 'function'
+            ? renderProps =>
+                `${defaultButtonClassName} ${className(renderProps)}`
+            : `${defaultButtonClassName} ${className || ''}`
         }
       >
-        {Icon && (
-          <Icon style={{ height: 15, paddingLeft: 5, paddingRight: 3 }} />
-        )}
```

Comment:
I think we should just remove this Icon since we can just pass the Icon as part of the `children` prop:

```jsx
<Button>
   <Icon />
    ....
</Button>
```

Reply:
We can definitely remove it now since it's not used.

The main reason for adding it as a prop instead of passing in `children` was the visual appearance. The icon didn't really look nice because of the large left margin the button automatically applies.